kvs-rest-clients/src/main/java/org/gbif/rest/client/RestClientFactory.java

```diff
@@ -0,0 +1,216 @@
+import org.springframework.cloud.openfeign.annotation.RequestHeaderParameterProcessor;
+import org.springframework.cloud.openfeign.annotation.RequestParamParameterProcessor;
+import org.springframework.cloud.openfeign.support.SpringMvcContract;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import feign.Feign;
+import feign.MethodMetadata;
+import feign.Util;
+import feign.form.spring.SpringFormEncoder;
+import feign.jackson.JacksonDecoder;
+
+import static feign.Util.checkState;
+import static feign.Util.emptyToNull;
+import static org.springframework.core.annotation.AnnotatedElementUtils.findMergedAnnotation;
+
+/**
+ * Factory class to create instances of the GBIF REST API clients.
+ * This is a simplified way of creating clients largely for test purposes.
+ * Clients can also be created using the ClientBuilder in the gbif-common-ws module
+ * which provides more configuration options.
+ */
+public class RestClientFactory {
+
+    private static final Integer DEFAULT_CONNECT_TIMEOUT_MILLISECONDS = 10_000;
+    private static final Integer DEFAULT_READ_TIMEOUT_MILLISECONDS = 60_000;
+
+    /**
+     * Creates a new instance of the NameUsageMatchService using the provided clientConfiguration.
+     * @param clientConfiguration the client configuration
+     * @return the NameUsageMatchService
+     */
+    public static NameUsageMatchingService createNameMatchService(ClientConfiguration clientConfiguration) {
+        return build(NameUsageMatchingService.class, clientConfiguration);
+    }
+
+    /**
+     * Creates a new instance of the GeocodeService using the provided clientConfiguration.
+     * @param clientConfiguration the client configuration
+     * @return the GeocodeService
+     */
+    public static GeocodeService createGeocodeService(ClientConfiguration clientConfiguration) {
+        return build(GeocodeService.class, clientConfiguration);
+    }
+
+    /**
+     * Creates a new instance of the GrscicollLookupService using the provided clientConfiguration.
+     * @param clientConfiguration the client configuration
+     * @return the GrscicollLookupService
+     */
+    public static GrscicollLookupService createGrscicollLookupService(ClientConfiguration clientConfiguration) {
+        return build(GrscicollLookupService.class, clientConfiguration);
+    }
+
+    /**
+     * Creates a new client instance.
+     */
+    private static <T> T build(Class<T> clazz, ClientConfiguration clientConfiguration) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(
+                new SimpleModule().addDeserializer(
+                        GeocodeResponse.class,
+                        new GeocodeResponse.GeocodeDeserializer()
+                )
+        );
+
+        Feign.Builder builder =
```

Comment:
we have this common class to create the feign clients https://github.com/gbif/gbif-common-ws/blob/74341a11b15cfd04516299aa9a3da8fd3acbf972/src/main/java/org/gbif/ws/client/ClientBuilder.java

would it be worth using it and if you use some things that are not supported adding them?

Reply:
I was trying to reduce dependencies. `gbif-common-ws` brings in a lot of dependencies.
That said, all this does work with the ClientBuilder, so if clients are already using gbif-common-ws, there isnt anything here that stops them using ClientBuilder instead.

Reply:
Ah ok, let's leave it here then not to fight a dependency hell.